### DISCUSSION TITLE
How to model a warm water tank?

### DISCUSSION THREAD

**Discussion by joppi588** [oemof/tespy#566] [Q&A] [answered]
Dear all,

I am seeking for advice on how to model a standard warm water tank such as
https://shwt-warmwasserspeicher.de/shwt_eingebettet/shop/ews8b1200500.png
I just want to model the heating of the water.

Physically, the tank consists of a heat exchanger, the water itself and the enclosing surface,
so I thought to set up a model like this (with two simple heat exchangers):
![grafik](https://github.com/user-attachments/assets/2445cf4e-ee4d-4693-aa71-e8a51ea69c15)

However I understand from the tutorials that pressure and mass flow are crucial model parameters;
I would need to include a pump for that, although in reality there is no water circulation in the tank.
Which approach should I take?

Unfortunately I could not find an answer in the examples and discussions.

Thanks,
Jochen

**Comment by fwitte** [comment]
Hi Jochen,

thank you for reaching out. I am out of the office at the moment. I will try to give you an answer not too far in the future. As a quick heads up: for these kinds of things I create two separate models. A tespy model for the stationary flow simulation and a separate storage model (which is not part of the tespy package, you have to code it by yourself or try to find some software for that). Then I couple the models by exchanging data between them, i.e. temperature or heat exchanged.

Best

Francesco

**Reply by joppi588** [reply]
I see. The heat exchanger in the sketch above would be replaced by a simple heat exchange _warm_tank_water_, with the ambient temperature being the current water temperature; the temperature drop at the tanks' surface would be calculated by a separate model.
That is, I would use two networks:

![grafik](https://github.com/user-attachments/assets/5dcce4ac-7f0e-4e2d-b298-201f609f0d81)



Thanks for the hint!

**Reply by fwitte** [reply, marked as answer]
Yes, that is the direction I was looking into.

I think you could also just use a single network, or one for charging and one for discharging. Then you can couple either of them to the storage model, that calculates thermal losses, change of internal temperature etc. :).

Have a nice day!
